pkg/storage/ingest/fetcher.go

```diff
@@ -466,14 +465,14 @@ func (g offsetRange) numOffsets() int64 {
 	return g.end - g.start
 }
 
-func findGapsInRecords(records []*kgo.Record, lastReturnedOffset int64) []offsetRange {
+func findGapsInRecords(records kgo.Fetches, lastReturnedOffset int64) []offsetRange {
 	var gaps []offsetRange
-	for _, r := range records {
-		if r.Offset != lastReturnedOffset+1 {
+	records.EachRecord(func(r *kgo.Record) {
+		if r.Offset > lastReturnedOffset+1 {
```

Comment:
`!=` --> `>`  because i saw a test panic that we're trying to report a gap that's negative in value (adding negative value to prometheus counter). I think the only way this can happen is if we re-consme some records. This isn't necessarily a problem, so I'm changing the comparison here to only record missing records, not records already consumed

Reply:
Would be nice to code this scenario in the `findGapsInRecords()` unit test, or at least leave a comment in the code.